### PR TITLE
Feature/notes support

### PR DESCRIPTION
**Pull request by scottfleming**
There was a bug processing the `notes` table from OMOP as follows:

Line [112](https://github.com/Medical-Event-Data-Standard/meds_etl/blob/8557dc3c13ff280c547146fb882949844b3d56cb/src/meds_etl/omop.py#L112) in meds_etl/src/meds_etl/omop.py defined a `source_concept_id` which in the case of the `notes` table would be `note_class_source_concept_id`. Key problem: `note_class_source_concept_id` is not a column in the `notes` table.
This is invoked in the definition of the `concept_id` column specification in line [118](https://github.com/Medical-Event-Data-Standard/meds_etl/blob/8557dc3c13ff280c547146fb882949844b3d56cb/src/meds_etl/omop.py#L118) and then again in the specification of `code` from line [126](https://github.com/Medical-Event-Data-Standard/meds_etl/blob/8557dc3c13ff280c547146fb882949844b3d56cb/src/meds_etl/omop.py#L126C24-L126C34) to try and map the `concept_id` column (which depends in part on the `note_class_source_concept_id` which does not exist) via the `concept_id_map`.
When the lazy polars dataframe is finally collected in line [221](https://github.com/Medical-Event-Data-Standard/meds_etl/blob/8557dc3c13ff280c547146fb882949844b3d56cb/src/meds_etl/omop.py#L221), it triggers a polars.exceptions.ColumnNotFoundError.

This bug fixes the error by first checking if `source_concept_id` exists, otherwise it falls back to the `concept_id` column.

Along the way, it also updates the progress monitoring to use `tqdm` which can be helpful for gauging how long the ETL has been running and (more importantly) estimating how long it will take to finish.

### CODE REVIEW

File: src/meds_etl/omop.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 import meds
 import polars as pl
 import pyarrow.parquet as pq
+from tqdm import tqdm
```
Comment (EthanSteinbergPrealize) [inline src/meds_etl/omop.py:15]: Can you add this to the dependencies in pyproject.toml?

File: src/meds_etl/omop.py [hunk 2]
```diff
@@ -258,38 +290,55 @@ def main():
...
             concept_name_map |= dict(zip(result["concept_id"], result["name"]))
 
+            # Assuming custom concepts have concept_id > 2000000000 we create a
+            # record for them in `code_metadata` with no parent codes. Such a
+            # custom code could be eg `STANFORD_RACE/Black or African American`
+            # with `concept_id` 2000039197
             custom_concepts = (
                 concept.filter(concept_id > 2_000_000_000)
-                .select(code=code, description=pl.col("concept_name"))
+                .select(concept_id=concept_id, code=code, description=pl.col("concept_name"))
```
Comment (EthanSteinbergPrealize) [inline src/meds_etl/omop.py:319]: Why are you pulling concept_id here?

**Review by EthanSteinbergPrealize — APPROVED** [review]
Looks good with two minor comments / questions!

File: src/meds_etl/omop.py (hunk 2 above)
Comment (scottfleming) [reply src/meds_etl/omop.py:401]: Makes debugging easier with a trivial amount of increased memory/time requirement, but I can remove if you prefer

File: src/meds_etl/omop.py (hunk 1 above)
Comment (scottfleming) [reply src/meds_etl/omop.py:15]: Will do!

File: src/meds_etl/omop.py (hunk 2 above)
Comment (EthanSteinbergPrealize) [reply src/meds_etl/omop.py:401]: That's fine. We can keep it. I was just wondering.

File: src/meds_etl/omop.py [hunk 3]
```diff
@@ -222,19 +251,99 @@ def transform_metadata(d):
...
 
+                # Write each shard to disk
                 for shard_index, shard in event_data.items():
                     fname = os.path.join(
                         temp_dir, str(shard_index), f'{table_name.replace("/", "_")}_{index}_{batch_index}_{i}.parquet'
                     )
                     shard.write_parquet(fname, compression="uncompressed")
 
 
+def process_shard(args: tuple[int, str, str]):
```
Comment (EthanSteinbergPrealize) [inline src/meds_etl/omop.py:262]: This is fine for now, but in the long term we really want to move this logic to the MEDS Flat ETL and have this code rely on the MEDS Flat ETL.

See https://github.com/Medical-Event-Data-Standard/meds_etl/pull/12

Comment (scottfleming) [reply src/meds_etl/omop.py:316]: Can delete this in a later PR. Prioritizing merge for now

Comment (EthanSteinbergPrealize) [reply src/meds_etl/omop.py:316]: Totally understandable. I'm fine with this being left here for now.

**Pull request merged**